tests/client/connstring.rs

```diff
@@ -0,0 +1,247 @@
+use mongodb::client::connstring;
+
+#[test]
+fn test_valid_uri() {
+    let valid_uris = vec!(
+        "mongodb://localhost",
+        "mongodb://localhost/",
+        "mongodb://localhost/?",
+        "mongodb://localhost:27017",
+        "mongodb://localhost:27017/",
+        "mongodb://localhost:27017/?",
+        "mongodb://127.0.0.1",
+        "mongodb://127.0.0.1/",
+        "mongodb://127.0.0.1/?",
+        "mongodb://127.0.0.1:27017",
+        "mongodb://127.0.0.1:27017/",
+        "mongodb://127.0.0.1:27017/?",
+        );
```

Comment:
I personally would have this `);` outdented to match up with the `let`, but that's subjective.